sormas-ui/src/main/java/de/symeda/sormas/ui/immunization/components/filter/ImmunizationFilterForm.java

```diff
@@ -0,0 +1,476 @@
+		}
+	}
+
+	@Override
+	protected Stream<Field> streamFieldsForEmptyCheck(CustomLayout layout) {
+		HorizontalLayout dateFilterLayout = (HorizontalLayout) getMoreFiltersContainer().getComponent(WEEK_AND_DATE_FILTER);
+		EpiWeekAndDateFilterComponent<ImmunizationDateType> weekAndDateFilter =
+			(EpiWeekAndDateFilterComponent<ImmunizationDateType>) dateFilterLayout.getComponent(0);
+
+		return super.streamFieldsForEmptyCheck(layout).filter(f -> f != weekAndDateFilter.getDateFilterOptionFilter());
+	}
+
+	private HorizontalLayout buildWeekAndDateFilter() {
+
+		EpiWeekAndDateFilterComponent<ImmunizationDateType> weekAndDateFilter = new EpiWeekAndDateFilterComponent<>(
+			false,
+			false,
+			null,
+			ImmunizationDateType.values(),
+			I18nProperties.getString(Strings.promptImmunizationDateType),
+			null,
+			this);
+		weekAndDateFilter.getWeekFromFilter().setInputPrompt(I18nProperties.getString(Strings.promptImmunizationEpiWeekFrom));
+		weekAndDateFilter.getWeekToFilter().setInputPrompt(I18nProperties.getString(Strings.promptImmunizationEpiWeekTo));
+		weekAndDateFilter.getDateFromFilter().setInputPrompt(I18nProperties.getString(Strings.promptImmunizationDateFrom));
+		weekAndDateFilter.getDateToFilter().setInputPrompt(I18nProperties.getString(Strings.promptImmunizationDateTo));
+
+		addApplyHandler(e -> onApplyClick(weekAndDateFilter));
+
+		HorizontalLayout dateFilterRowLayout = new HorizontalLayout();
+		dateFilterRowLayout.setSpacing(true);
+		dateFilterRowLayout.setSizeUndefined();
+
+		dateFilterRowLayout.addComponent(weekAndDateFilter);
+
+		return dateFilterRowLayout;
+	}
+
+	private void onApplyClick(EpiWeekAndDateFilterComponent<ImmunizationDateType> weekAndDateFilter) {
+		ImmunizationCriteria criteria = getValue();
+
+		DateFilterOption dateFilterOption = (DateFilterOption) weekAndDateFilter.getDateFilterOptionFilter().getValue();
+		Date fromDate, toDate;
+		if (dateFilterOption == DateFilterOption.DATE) {
+			Date dateFrom = weekAndDateFilter.getDateFromFilter().getValue();
+			fromDate = dateFrom != null ? DateHelper.getStartOfDay(dateFrom) : null;
+			Date dateTo = weekAndDateFilter.getDateToFilter().getValue();
+			toDate = dateFrom != null ? DateHelper.getEndOfDay(dateTo) : null;
+		} else {
+			fromDate = DateHelper.getEpiWeekStart((EpiWeek) weekAndDateFilter.getWeekFromFilter().getValue());
+			toDate = DateHelper.getEpiWeekEnd((EpiWeek) weekAndDateFilter.getWeekToFilter().getValue());
+		}
+		if ((fromDate != null && toDate != null) || (fromDate == null && toDate == null)) {
+			criteria.setDateFilterOption(dateFilterOption);
+			ImmunizationDateType immunizationDateType = (ImmunizationDateType) weekAndDateFilter.getDateTypeSelector().getValue();
+			criteria.setImmunizationDateType(immunizationDateType);
+			criteria.setFromDate(fromDate);
+			criteria.setToDate(toDate);
+		} else {
+			setNotifications(dateFilterOption);
+		}
+	}
+
+	public static void setNotifications(DateFilterOption dateFilterOption) {
```

Comment:
Move this to a general helper class please, since it's also called in other filter forms